src/structures/GuildChannel.js

```diff
@@ -571,7 +571,10 @@ class GuildChannel extends Channel {
    * @readonly
    */
   get deletable() {
-    return this.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_CHANNELS, false);
+    return (
+        this.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_CHANNELS, false) && 
+        (this.guild.rulesChannelID !== this.id || this.guild.systemChannelID !== this.id)
```

Comment:
the guild system channel is deletable, its only the rules channel that you can't delete

Reply:
the `publicUpdatesChannelID` also can't be deleted